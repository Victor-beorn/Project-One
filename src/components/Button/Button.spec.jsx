import { render, screen } from '@testing-library/react';
import { Button } from '.';
import userEvent from '@testing-library/user-event';

describe('<Button/>', () => {
  it('should render the button with the text', () => {
    render(<Button text="Carregar mais posts" />);
    expect.assertions(1);

    const button = screen.getByRole('button', { name: /Carregar mais posts/i });
    expect(button).toBeInTheDocument();
  });

  it('should call function on button click', () => {
    const fn = jest.fn();
    render(<Button text="Carregar mais posts" onClick={fn} />);

    const button = screen.getByRole('button', { name: /Carregar mais posts/i });

    userEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should be disable when disable is true', () => {
    render(<Button text="Carregar mais posts" disable={true} />);

    const button = screen.getByRole('button', { name: /Carregar mais posts/i });

    expect(button).toBeDisabled();
  });

  it('should match snapshot', () => {
    const { container } = render(<Button text="Carregar mais posts" disable={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
