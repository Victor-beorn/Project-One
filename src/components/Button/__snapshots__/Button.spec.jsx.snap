// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should match snapshot 1`] = `
<button
  class="button"
  disabled=""
>
  Carregar mais posts
</button>
`;
